fleet/redis/consumer/streams.go

```diff
@@ -108,6 +123,68 @@ func (es eventStore) handleDatasetCreate(ctx context.Context, e createDatasetEve
 	return es.commsService.NotifyGroupNewAgentPolicy(ctx, ag, e.policyID, e.ownerID)
 }
 
+func decodePolicyUpdate(event map[string]interface{}) (updatePolicyEvent, error) {
+	val := updatePolicyEvent{
+		id:      read(event, "id", ""),
+		ownerID: read(event, "owner_id", ""),
+	}
+
+	strgroups := read(event, "groups_ids", "")
+	val.groupsIDs = strings.Split(strgroups, ",")
+
+	strpolicy := read(event, "policy", "")
+	var policy types.Metadata
+	if err := json.Unmarshal([]byte(strpolicy), &policy); err != nil {
+		return updatePolicyEvent{}, err
+	}
+
+	return val, nil
+}
+
+// the policy service is notifying that a policy has been updated
+// notify all agents in the AgentGroup specified in the dataset about the policy update
+func (es eventStore) handlePolicyUpdate(ctx context.Context, e updatePolicyEvent) error {
+	// todo check if it will be necessary create a new comms function to notify a update
```

Comment:
this is actually a good question - I think the way you have it might be okay though (using NotifyGroupNewAgentPolicy)

Reply:
Nice